### PR TITLE
Add tables in liquibase to back spring session

### PR DESCRIPTION
**Pull request by emmastephenson**
## Related Issue or Background Info

#1402 

## Changes Proposed

- Add tables in our database to back spring session.

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support

### CODE REVIEW

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -1758,3 +1758,75 @@ databaseChangeLog:
...
+      author: emma@skylight.digital
+      comment: Add a table to back spring sessions.
+      changes: 
+        - createTable:
+              tableName: spring_session
+              remarks: Data store for user sessions, as created by SpringSession.
+              columns:
+                - column:
+                    name: primary_id
+                    type: *idtype
```
Comment (benwarfield-usds) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:1772]: Canonically this is a `char(36)`, not a UUID—is Spring Session actually using a UUID, and they just haven't updated their example DDL to reflect postgresql having a UUID type? 🤣

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -1758,3 +1758,75 @@ databaseChangeLog:
...
+                    remarks: The name or UUID of the user for this session.
+        - createTable:
+              tableName: spring_session_attributes
+              remarks: Attribute store for user sessions.
+              columns:
+                - column:
+                    name: session_primary_id
+                    type: *idtype
+                    remarks: The immutable internal key to identify this session.
+                    constraints:
```
Comment (benwarfield-usds) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:1819]: foreign key here?

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 3]
```diff
@@ -1758,3 +1758,75 @@ databaseChangeLog:
...
+                    type: DATETIME
+                    remarks: The time this session will expire (last_access_time + max_inactive_interval.)
+                    constraints: 
+                      nullable: false
+                - column:
+                    name: principal_name
+                    type: *string
+                    remarks: The name or UUID of the user for this session.
+        - createTable:
+              tableName: spring_session_attributes
```
Comment (benwarfield-usds) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:1812]: I believe this table has a composite primary key of `session_primary_id, attribute_name`.

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -1758,3 +1758,75 @@ databaseChangeLog:
...
+        - createTable:
+              tableName: spring_session
+              remarks: Data store for user sessions, as created by SpringSession.
+              columns:
+                - column:
+                    name: primary_id
+                    type: *idtype
+                    remarks: The immutable internal key to identify this session.
+                    constraints:
+                      nullable: false
```
Comment (benwarfield-usds) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:1775]: Isn't this the primary key?

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 1 above)
Comment (emmastephenson) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml]: Sadly not! At least not from [this example](https://github.com/spring-projects/spring-session/blob/main/spring-session-jdbc/src/main/resources/org/springframework/session/jdbc/schema-postgresql.sql). I updated to a `char(36)` instead 😃

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 4 above)
Comment (emmastephenson) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:1775]: Yes, good call - updated. (I also added updated the rest of this file to match that postgres example I linked above, which included adding some indexes and more primary/foreign keys)

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 3 above)
Comment (emmastephenson) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:1827]: Good catch, added!

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 2 above)
Comment (emmastephenson) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml]: Yup 👍

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -1758,3 +1758,96 @@ databaseChangeLog:
...
+                    type: DATETIME
+                    remarks: The last time the session id was used.
+                    constraints:
+                      nullable: false 
+                - column:
+                    name: max_inactive_interval
+                    type: int
+                    remarks: The maximum inactive interval time before a session times out and is invalidated.
+                    constraints: 
+                      nullable: facilities
```
Comment (benwarfield-usds) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:1803]: ?

**Review by benwarfield-usds — APPROVED** [review]
I'm a bit suspicious of one YAML entry being an incorrect autocompletion, otherwise LGTM.

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 5 above)
Comment (emmastephenson) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml]: Definitely an incorrect autocomplete 🤦  thanks for catching that!

**Pull request merged**